### PR TITLE
Named nodes

### PR DESCRIPTION
**Pull request by Jegp**
Replaces indices with string names

### CODE REVIEW

File: nir/ir.py [hunk 1]
```diff
@@ -100,7 +109,7 @@ class CubaLIF(NIRNode):
     r: np.ndarray  # Resistance
     v_leak: np.ndarray  # Leak voltage
     v_threshold: np.ndarray  # Firing threshold
-    w_in: np.ndarray = 1.  # Input current weight
+    w_in: np.ndarray = 1.0  # Input current weight
```
Comment (matjobst) [inline nir/ir.py:112]: This default value should be an np array.

File: tests/test_readwrite.py [hunk 2]
```diff
@@ -27,121 +27,101 @@ def factory_test_graph(ir: nir.NIRGraph):
 
 
 def test_simple():
-    ir = nir.NIRGraph(nodes=[nir.Affine(weight=[1, 2, 3], bias=4)], edges=[(0, 0)])
+    ir = nir.NIRGraph(
+        nodes={"a": nir.Affine(weight=[1, 2, 3], bias=4)}, edges=[("a", "b")]
```
Comment (matjobst) [inline tests/test_readwrite.py:31]: There is no node "b", so this test should be failing!

File: nir/ir.py [hunk 3]
```diff
@@ -21,9 +21,17 @@ class NIRGraph(NIRNode):
 
     A graph of computational nodes and identity edges."""
 
-    nodes: typing.List[NIRNode]  # List of computational nodes
+    nodes: typing.Dict[str, NIRNode]  # List of computational nodes
     edges: Edges
 
+    @staticmethod
+    def from_list(*nodes: NIRNode) -> "NIRGraph":
+        """Create a sequential graph from a list of nodes by labelling them after indices."""
```
[the remark this answers is not in the document]
Comment (Jegp) [reply nir/ir.py]: That's a good point! Would it be ok to defer that for now?

[the remark this answers is not in the document]
Comment (matjobst) [reply nir/ir.py]: Sure, I will create an issue!

File: tests/test_readwrite.py (hunk 2 above)
Comment (Jegp) [reply tests/test_readwrite.py:31]: Hehe, true!

File: tests/test_ir.py [hunk 4]
```diff
@@ -1,93 +1,100 @@
...
+    assert ir.nodes["in"].shape == [
         3,
     ]
-    assert ir.nodes[1].delay == [1, 2, 3]
-    assert ir.edges == [(0, 1), (1, 2)]
+    assert ir.nodes["d"].delay == [1, 2, 3]
+    assert ir.edges == [("in", "d"), ("d", "out")]
+
+
+def test_cuba_lif():
```
Comment (matjobst) [inline tests/test_ir.py:72]: This test does not test anything.
Additionally, the Ruff complains about the variable name 'l' which might be ambiguous with 1.

Comment (Jegp) [reply tests/test_ir.py:72]: Good catch. Sorry for the incomplete test.

File: nir/ir.py (hunk 1 above)
Comment (Jegp) [reply nir/ir.py:112]: Addressed in #21

**Comment by biphasic** [comment]
LGTM

**Pull request merged**